### PR TITLE
API Docs Overhaul

### PR DESCRIPTION
**Pull request by bengl**
### What does this PR do?
- clarify some things in TS
- move orphanable from TracerOptions to TracerOptions
- include SpanOptions in docs
- document format strings

### Motivation

Make the docs more usable and correct.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -3,14 +3,16 @@ import { LookupFunction } from 'net';
 import * as opentracing from "opentracing";
 import { SpanOptions } from "opentracing/lib/tracer";
 
-export { SpanOptions };
```
Comment (rochdev) [inline index.d.ts:6]: Why are these no longer exported?

File: docs/tsconfig.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "baseUrl": "."
   },
   "files": [
-    "../index.d.ts"
+    "../index.d.ts",
+    "../node_modules/opentracing/lib/tracer.d.ts",
```
Comment (rochdev) [inline docs/tsconfig.json:10]: Why is this needed? How does it change the output?

**Comment by bengl** [comment]
This was a bit aspirational. TS is more complicated that it may seem.
